tests/unit/test_remote/test_localMachine.py

```diff
@@ -0,0 +1,89 @@
+# A part of NonVisual Desktop Access (NVDA)
+# Copyright (C) 2025 NV Access Limited
+# This file may be used under the terms of the GNU General Public License, version 2 or later, as modified by the NVDA license.
+# For full terms and any additional permissions, see the NVDA license file: https://github.com/nvaccess/nvda/blob/master/copying.txt
+
+from functools import reduce
+from itertools import product
+from typing import Any, NamedTuple
+import unittest
+from unittest import mock
+import winreg
+from _remoteClient.localMachine import LocalMachine, SoftwareSASGeneration
+
+
+class CanSendSASFactors(NamedTuple):
+	"""Possible external factors that could influence whether NVDA can call SendSAS."""
+
+	hasUIAccess: bool
+	openKeyRaises: Exception | None
+	queryValueRaises: Exception | None
+	valueData: Any
+	valueType: int
+	isRunningOnSecureDesktop: bool
+
+
+FACTOR_VALUES = {
+	"hasUIAccess": (True, False),
+	"openKeyRaises": (None, FileNotFoundError, OSError),
+	"queryValueRaises": (None, FileNotFoundError),
+	"valueData": (*SoftwareSASGeneration, 999, "nonsense"),
+	"valueType": (winreg.REG_DWORD, 999),
+	"isRunningOnSecureDesktop": (True, False),
+}
+"""Possible values for each of the factors in :class:`CanSendSASFactors`."""
+
+EXPECTED_SUCCESS_CASES = (
+	CanSendSASFactors(True, None, None, SoftwareSASGeneration.UIACCESS, winreg.REG_DWORD, False),
```

Comment:
Can you please add the keywords to these to make them more readable. Otherwise you have to keep referring back to the data type e.g.
```suggestion
	CanSendSASFactors(
		hasUIAccess=True,
		openKeyRaises=None,
		queryValueRaises=None,
		valueData=SoftwareSASGeneration.UIACCESS,
		valueType=winreg.REG_DWORD,
		isRunningOnSecureDesktop=False
	),
```